controllers/periodic_healthcheck.go

```diff
@@ -138,21 +143,13 @@ func (r *EtcdadmClusterReconciler) periodicEtcdMembersHealthCheck(ctx context.Co
 		}
 	}
 
-	ep, err := NewEtcdPlane(ctx, r.Client, currClusterHFConfig.cluster, etcdCluster, currClusterHFConfig.ownedMachines)
-	if err != nil {
-		return errors.Wrap(err, "Error initializing internal object EtcdPlane")
-	}
-
 	var retErr error
 	for machineEndpoint, machineToDelete := range currClusterHFConfig.unhealthyMembersToRemove {
-		// if user delete Machine CR, removeEtcdMember won't be needed
-		if machineToDelete != nil {
-			if err := r.removeEtcdMember(ctx, etcdCluster, cluster, ep, machineToDelete); err != nil {
-				// log and save error and continue deletion of other members, deletion of this member will be retried since it's still part of unhealthyMembersToRemove
-				r.Log.Error(err, fmt.Sprintf("error removing etcd member machine %v", machineToDelete.Name))
-				retErr = multierror.Append(retErr, err)
-				continue
-			}
+		if err := r.removeEtcdMachine(ctx, etcdCluster, cluster, machineToDelete, getEtcdMachineAddressFromClientURL(machineEndpoint)); err != nil {
+			// log and save error and continue deletion of other members, deletion of this member will be retried since it's still part of unhealthyMembersToRemove
```

Comment:
I know that this code was already here but: does that mean this loop can keep iterating forever if a particular machine keeps erroring out?

Reply:
for loop iteration will happen once every time the periodic healthcheck loop runs if that's what you're asking